core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1475,41 +1507,55 @@ ClientTransport obtainActiveTransport() {
     @Override
     public void shutdown() {
       logWarningIfNotInSyncContext("Subchannel.shutdown()");
-      synchronized (shutdownLock) {
-        if (shutdownRequested) {
-          if (terminating && delayedShutdownTask != null) {
-            // shutdown() was previously called when terminating == false, thus a delayed shutdown()
-            // was scheduled.  Now since terminating == true, We should expedite the shutdown.
-            delayedShutdownTask.cancel(false);
-            delayedShutdownTask = null;
-            // Will fall through to the subchannel.shutdown() at the end.
-          } else {
-            return;
+      syncContext.execute(new Runnable() {
+          @Override
+          public void run() {
+            internalShutdown();
           }
+        });
+    }
+
+    private void internalShutdown() {
```

Comment:
This method is only used in one place. Consider move it to a named inner class?

Reply:
This method is intended to replace `shutdown()` when we are ready to turn the warning into an exception. I have added a comment to make that clear.